imsim/instcat.py

```diff
@@ -274,10 +286,23 @@ def getSED(self, index):
             sed = sed.withMagnitude(0, self._bp500)  # Normalize to mag 0
             self._sed_cache[name] = sed
 
-        # TODO: Handle the dust effects.  internal, then redshift, then galactic.
-        #       Also probably need to think about the normalization and how that interacts with
-        #       the flux and/or mag.
-        return sed.atRedshift(redshift)
+        iAv, iRv, mwAv, mwRv = self.getDust(index)
+
+        # TODO: apply internal extinction here
+
+        # Apply redshift.
+        sed = sed.atRedshift(redshift)
+
+        # Apply Milky Way extinction.
+        extinction = F19(Rv=mwRv)
+        wl = np.linspace(300, 1200, 901)
+        ext = extinction.extinguish(wl*u.nm, Av=mwAv)
+        spec = galsim.LookupTable(wl, ext)
+        mw_ext = galsim.SED(spec, wave_type='nm', flux_type='1')
+        sed = sed*mw_ext
+        sed = sed.thin()
```

Comment:
I mentioned this before.  Don't do this.  It's super slow.  Instead, use `wl=sed.wave_list` rather than make a new linspace.

Reply:
right, sorry, I fixed it in the skycat.py module, but neglected to fix it here.